### PR TITLE
upgrade lucene to r1702265

### PR DESCRIPTION
**Pull request by rmuir**
Mainly motivated to remove sun.nio.ch permission. But it has some boostings changes too.

### CODE REVIEW

File: update_lucene.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+perl dev-tools/src/main/resources/license-check/check_license_and_sha.pl \
+     --update plugins/analysis-icu/licenses/ plugins/analysis-icu/target/releases/analysis-icu-3.0.0-SNAPSHOT.zip analysis-icu-3.0.0-SNAPSHOT
+perl dev-tools/src/main/resources/license-check/check_license_and_sha.pl \
+     --update plugins/analysis-kuromoji/licenses/ plugins/analysis-kuromoji/target/releases/analysis-kuromoji-3.0.0-SNAPSHOT.zip analysis-kuromoji-3.0.0-SNAPSHOT
+perl dev-tools/src/main/resources/license-check/check_license_and_sha.pl \
+     --update plugins/analysis-phonetic/licenses/ plugins/analysis-phonetic/target/releases/analysis-phonetic-3.0.0-SNAPSHOT.zip analysis-phonetic-3.0.0-SNAPSHOT
+perl dev-tools/src/main/resources/license-check/check_license_and_sha.pl \
+     --update plugins/analysis-smartcn/licenses/ plugins/analysis-smartcn/target/releases/analysis-smartcn-3.0.0-SNAPSHOT.zip analysis-smartcn-3.0.0-SNAPSHOT
+perl dev-tools/src/main/resources/license-check/check_license_and_sha.pl \
+     --update plugins/analysis-stempel/licenses/ plugins/analysis-stempel/target/releases/analysis-stempel-3.0.0-SNAPSHOT.zip analysis-stempel-3.0.0-SNAPSHOT
```
Comment (jpountz) [inline update_lucene.sh]: woohoo!

**Comment by jpountz** [comment]
Maybe we could mark these super super slow tests with `@Nightly` instead of disabling them entirely? Regarding the thread dumps I think @jasontedor added this feature in order to have more information when a hard-to-reproduce test fails, but I tend to agree it makes the output very verbose given the number of threads elasticsearch starts. I don't know how useful it proved, @jasontedor what do you think?

**Comment by rmuir** [comment]
Feel free to push changes. I am done here. 

I just want to add the security permission. I do not care about the crappy tests

**Comment by jpountz** [comment]
OK, I'll take care of addressing potential comments and merging this PR.

**Comment by jasontedor** [comment]
I think it can be removed. I filed #13440 to remove it.

**Comment by jpountz** [comment]
thanks @jasontedor !

**Comment by jpountz** [comment]
I rebased to include @jasontedor 's changes and fixed the remaining bugs. It should be good now.

**Comment by rmuir** [comment]
looks great

**Comment by rmuir** [comment]
This branch now works with jigsaw (`mvn verify` passes). I wrote up a summary here: http://mail.openjdk.java.net/pipermail/jigsaw-dev/2015-September/004511.html

**Comment by rmuir** [comment]
I just backported this to 2.x

**Pull request merged**